helm/idol-library/templates/_ingress.tpl

```diff
@@ -27,6 +27,19 @@ metadata:
       "nginx.ingress.kubernetes.io/rewrite-target" "/$1"
       "nginx.ingress.kubernetes.io/backend-protocol" (ternary "HTTPS" "HTTP" $component.usingTLS)
   ) -}}
+  {{- if and $ingress.exposeIndexPort $ingress.exposeServicePort }}
+    {{- $_ := set $annotations "nginx.ingress.kubernetes.io/configuration-snippet" `more_set_headers "AciPortPath: /content/";
+more_set_headers "IndexPortPath: /index/";
+more_set_headers "ServicePortPath: /content/service/";` -}}
+  {{- else if $ingress.exposeIndexPort }}
+    {{- $_ := set $annotations "nginx.ingress.kubernetes.io/configuration-snippet" `more_set_headers "AciPortPath: /content/";
+more_set_headers "IndexPortPath: /index/";` -}}
+  {{- else if $ingress.exposeServicePort }}
+    {{- $_ := set $annotations "nginx.ingress.kubernetes.io/configuration-snippet" `more_set_headers "AciPortPath: /content/";
+more_set_headers "ServicePortPath: /content/service/";` -}}
+  {{- else }}
+    {{- $_ := set $annotations "nginx.ingress.kubernetes.io/configuration-snippet" `more_set_headers "AciPortPath: /content/";` -}}
+  {{- end -}}
```

Comment:
This should use the provided values for `ingress.path`, `ingress.indexPath` and `ingress.servicePath`.

Could probably tidy up the logic by using another helm variable to build the snippet value:
```
{{- $headers := print 
    (ternary "" (printf "more_set_headers \"AciPortPath: %s\";\n" $ingress.path)  (empty $ingress.path))
    (ternary "" (printf "more_set_headers \"ServicePortPath: %s\";\n" $ingress.servicePath)  (empty $ingress.servicePath))
    (ternary "" (printf "more_set_headers \"IndexPortPath: %s\";\n" $ingress.indexPath)  (empty $ingress.indexPath))
-}}
```

We don't need the `.exposeBlah` variables if we just check if the supplied path is empty